ratis-common/src/main/java/org/apache/ratis/util/FileUtils.java

```diff
@@ -85,6 +85,43 @@ static void deleteFile(File f) throws IOException {
     delete(f.toPath());
   }
 
+  /**
+   * Moves the directory. If any file is locked, the exception is caught
+   * and logged and continues to other files.
+   * @param source
+   * @param dest
+   * @throws IOException
+   */
+  static void moveDirectory(Path source, Path dest) throws IOException {
+    LOG.trace("moveDirectory source: {} dest: {}", source, dest);
+    createDirectories(dest);
+    Files.walkFileTree(source, new SimpleFileVisitor<Path>() {
+      @Override
+      public FileVisitResult preVisitDirectory(Path dir,
+          BasicFileAttributes attrs) throws IOException {
+        Path targetPath = dest.resolve(source.relativize(dir));
+        if (!Files.exists(targetPath)) {
+          createDirectories(targetPath);
+        }
```

Comment:
Since we are making doing createDirectories call for dest, I do not think we need this.

Reply:
This call is required for sub folders, otherwise only the files get copied.

Reply:
This call is required for creating sub folders, otherwise only parent files get copied.